### ISSUE TITLE
Calling ProjectTo causes InvalidOperationException when the composite foreign key set

### ISSUE THREAD

**Issue by sadedil** [AutoMapper/AutoMapper#3353]
### Source/destination types

```csharp
Product -> ProductDto
ProductCategory -> ProductCategoryDto
```

### Mapping configuration

```csharp  private static readonly MapperConfiguration _mapperConfiguration = new MapperConfiguration(cfg =>
  {
    cfg.CreateMap<Product, ProductDto>();
    cfg.CreateMap<ProductCategory, ProductCategoryDto>();
  });
```

### Expected behavior

I expect, execute this lines without problem when composite FK is set
```csharp
  var query = context.Products;
  var products = await query.ProjectTo<ProductDto>(_mapperConfiguration).ToListAsync();
```

### How I set the composite foreign key

With this last two lines: 
```csharp
  modelBuilder.Entity<Product>()
    .HasOne(p => p.ProductCategory)
    .WithMany(pc => pc.Products)
    .HasForeignKey(p => new { p.TenantId, p.ProductCategoryId })
    .HasPrincipalKey(pc => new { pc.TenantId, pc.ProductCategoryId });
```

### Actual behavior

Getting this error:
```
Unhandled Exception: System.InvalidOperationException: Nullable object must have a value.
   at lambda_method(Closure , Object[] )
   at Microsoft.EntityFrameworkCore.Query.ExpressionVisitors.Internal.TaskLiftingExpressionVisitor._ExecuteAsync[T](IReadOnlyList`1 taskFactories, Func`2 selector)
   at Microsoft.EntityFrameworkCore.Query.Internal.AsyncLinqOperatorProvider.AsyncSelectEnumerable`2.AsyncSelectEnumerator.MoveNext(CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Query.Internal.AsyncLinqOperatorProvider.ExceptionInterceptor`1.EnumeratorExceptionInterceptor.MoveNext(CancellationToken cancellationToken)
   at System.Linq.AsyncEnumerable.Aggregate_[TSource,TAccumulate,TResult](IAsyncEnumerable`1 source, TAccumulate seed, Func`3 accumulator, Func`2 resultSelector, CancellationToken cancellationToken) in D:\a\1\s\Ix.NET\Source\System.Interactive.Async\Aggregate.cs:line 120
   at CompositeNullableForeignKeySample.Program.Main(String[] args) in C:\Users\mustafa.sadedil\source\repos\CompositeNullableForeignKeySample\Program.cs:line 113
   at CompositeNullableForeignKeySample.Program.<Main>(String[] args)
```

### Steps to reproduce

I've created a public repo to reproduce this scenario. You can find the working and non-working versions (and more detailed descriptions) there:

https://github.com/sadedil/CompositeNullableForeignKeySample

**Comment by lbargaoanu** [comment]
That looks like an EF Core issue. Check [the execution plan](https://docs.automapper.org/en/latest/Understanding-your-mapping.html). Run that as a LINQ statement, without AM. You should get the same result.

**Comment by sadedil** [comment]
Hİ @lbargaoanu,

Actually it works very well if I use simple LINQ or just use Map method instead of ProjectTo. Did you try my sample repo?

By the way, I've checked the execution plan with or without composite key and it is same in both situation.

**Comment by sadedil** [comment]
Hi @lbargaoanu,

Is this issue will be stay closed? Do you have an SLA or something? Of course you can choose to not to do anything. But I think closing an (isolated and reproduced) issue just with a guess, is not proper way of communication and issue tracking.

**Comment by lbargaoanu** [comment]
You saw the execution plan. What _exactly_ is wrong with it?

**Comment by sadedil** [comment]
`ProjectTo` generates expression like below (I've simplified to reduce line count). And this expression throws `System.InvalidOperationException: 'Nullable object must have a value.'` when I set composite principal key to `Product` like this `.HasPrincipalKey(pc => new { pc.TenantId, pc.ProductCategoryId })`

```csharp
var products = context.Products.Select(
	dtoProduct => new ProductDto
	{
		ProductName = dtoProduct.ProductName,
		ProductCategory = (dtoProduct.ProductCategory == null)
		? null
		: new ProductCategoryDto {
			ProductCategoryId = dtoProduct.ProductCategory.ProductCategoryId,
			ProductCategoryName = dtoProduct.ProductCategory.ProductCategoryName}
	}).ToList();
```

But when I change the null check from `dtoProduct.ProductCategory == null` to  `dtoProduct.ProductCategory == null || dtoProduct.ProductCategory.ProductCategoryId == null`  manually it's started to working. (Of course I can not do this when I use `ProjectTo`)

I'm not sure the `ProjectTo` should generate expression that can be supported by EF Core?

**Solutions (which is not working for us):**
- Don't use composite `HasPrincipalKey` and `HasForeignKey` (but we are going with Code First)
- Don't use `ProjectTo`: Our project heavily rely on this.

So, I'm stuck.
